### DISCUSSION TITLE
Node Red eines anderen Rechners (Home Assistant) verwenden um SAE Werte abzurufen

### DISCUSSION THREAD

**Discussion by ToniBecause** [camueller/SmartApplianceEnabler#502] [Q&A]
Hallo, 
ich möchte die gemessenen Stromverbruachswerte meiner S0 Zähler von SAE V2.2.1 in meinem Home Assistant weiterverwenden. Zuerst dachte ich ich könnte das Dashboard welches via Node Red von SAE bereitgestellt wird, via scraping zu abzufragen, leider erfolglos.

Gestern kam mir dann folgende Idee:
Ich könnte ja via NodeRED und dem MQTT Broker von meinem Home Assistant Raspberry Pi auf den SAE Raspberry PI zugreifen und dann dort über einen Sensor die Daten im Home Assistant weiter zu verarbeiten.
Ich habe somit den Flow von Alex in dem NodeRED meiner Raspberry Pi auf dem Home Assistant läuft 1:1 nachgebaut. Beim MQTT Broker habe ich lediglich die IP von 127.0.0.1 auf die IP meines Raspberr Pi´s auf dem SAE läuft geändert.
Leider kommt bei der Flow Ansicht unter dem MQTT Broker immer nur "Verbindung wird aufgebaut" anstatt "verbunden".
Ich habe nun keine Idee wo ich suchen müsste. Evtl muss ich auf meinem Home Assistant Rechner nur irgendwas nachinstallieren?
Vielleicht hat hier ja jemand einen Tip wonach ich schauen könnte, denn via Debug sehe ich nichts im NodeRED.

Geht das überhaupt von einem anderem Rechner auf SAE zuzugreifen? Ich muss leider so blöd nachfragen da ich das ganze Konstrukt noch nicht so ganz verinnerlicht habe und mit NodeRED sowie MQTT noch null Berührungspunkte hatte.

**Comment by camueller** [comment]
SAE, MQTT-Broker und NodeRED können selbstverständlich auf verschiedenen Hosts laufen. Bei Dir ist also irgendetwas nicht richtig konfiguriert.

Ich bin immer wieder verwundert, warum User oft komplexe Konfigurationen anstreben, ohne das notwendige Wissen dafür zu haben. Man kauft sich ja mit 18 Jahren auch nicht gleich einen Rennwagen, sondern fängt mit einem normalen Auto an.

**Comment by ToniBecause** [comment]
Komplex würde ich es nicht unbedingt nennen, aber sei es drum. Die SMA Wallbox zu steuern bzw Funktionalitäten über Home Assist zu erstellen wovor SMA sich seit 2 Jahren wehement wehrt fand ich herausvordender ;-)

Wenn aber meine Idee grundsätzlich funktionieren solle passt das schon mal. Vielleicht findet sich hier jemand der ähnliches schon mal gemacht hat. 
Eben hatte ich mal kurz Zeit gehabt und beim Mqtt Broker Dienst des Home Assistant die URL und Port vom SAE MQTT Host angegeben ohne User und Passwort und auch dort konnte keine Verbindung aufgebaut werden. Ich muss also am SAE Host mal nach den Einstellungen schauen. Wenn ich es gelöst habe melde ich mich wieder.

**Comment by ToniBecause** [comment]
Aufgrund eines anderen Projektes bei dem MQTT wunderbar funktioniert muss ich hier nochmal nachhaken. Wenn ich in der Konfigurationseite von SAE von Mqtt die IP 127.0.0.1 auf die IP von SAE änder geht gar nichts mehr. Eigentlich sollte da doch kein Unterschied sein,oder?. Wo kann man denn den Mqtt Dienst vom SAE konfigurieren?

**Comment by camueller** [comment]
> Wo kann man denn den Mqtt Dienst vom SAE konfigurieren?

Siehe https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/InstallationManual_DE.md#mqtt-broker
